filament/backend/src/opengl/platforms/PlatformEGLWayland.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TNT_FILAMENT_DRIVER_OPENGL_PLATFORM_EGL_WAYLAND_H
+#define TNT_FILAMENT_DRIVER_OPENGL_PLATFORM_EGL_WAYLAND_H
+
+#include <cstdint>
+
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <wayland-egl.h>
+
+#include <backend/DriverEnums.h>
+
+#include "private/backend/OpenGLPlatform.h"
+
+namespace filament {
+
+class PlatformEGLWayland : public backend::OpenGLPlatform {
```

Comment:
This should derive from `PlatformEGL`, there should be a lot of code that can be shared between both.
Is there/what are the limitations in `PlatformEGL` that prevent it from being used here?

Reply:
Okay, so in that case, I think it is better to pull out all the GL/EGL changes from this PR, and keep only the Vulkan ones.  We can discuss/work on the GL/EGL stuff separately, in another PR. It's better to not tie them together.
Thanks.

Reply:
BTW, Is Wayland guaranteed to have `EGL_KHR_no_config_context`? If so, we don't need the "dummy" pbuffer and we could change EGLPlatform to skip creating the pbuffer in that case.

Reply:
@pixelflinger Sounds good.  Will do.

Wayland does not guarantee `EGL_KHR_no_config_context`.  The source shows it is just checking if it's present, so whatever the driver provides.  I'm finding it not present on a few machines.
https://github.com/wayland-project/weston/search?q=EGL_KHR_no_config_context

Reply:
What is odd is that I'm running fairly Mesa drivers from last two releases and PlatformEGL is not detecting the `EGL_KHR_no_config_context` extension.  I would expect it should be present.  I'll dig into that a bit more prior to pulling it.

Reply:
I played around a bit extending PlatformEGL.  Passing `EGL_RECORDABLE_ANDROID` generates `EGL_BAD_CONFIG` error, and it exits.  Wrapping line 113 with `#if defined(__ANDROID__)` I moved on.  Commenting out eglCreatePbufferSurface (mEGLDummySurface == nullptr) context is created and function returns.  

Later hitting 
```
in static filament::backend::Driver *filament::OpenGLDriver::create(filament::backend::OpenGLPlatform *const, void *const):135
in file /b/toyota/filament/filament/backend/src/opengl/OpenGLDriver.cpp
reason: OpenGL ES 3.0 minimum needed (current 0.0)
```

What about the primary EGL surface created in initial SDL create window call?  This EGL surface pointer is only exposed in SDL >= 2.0.15.

Reply:
@pixelflinger I backed out Wayland EGL.